server/src/data_store.cpp

```diff
@@ -198,4 +224,38 @@ int DataStore::DbCallback(void* not_used, int argc, char** argv, char** azColNam
     return 0;
 }
 
+bool DataStore::ReadDistinctMacAddrs(const std::string& device_id, std::vector<std::string>& mac_addrs)
+{
+    console_->debug("+ DataStore::GetRSSIDataStream");
+
+    bool        result = false;
+    std::string sql    = "SELECT DISTINCT mac_addr FROM  dev_" + device_id + ";";
+
+    sqlite3_stmt* selectStmt;
+    sqlite3_prepare(database_, sql.c_str(), static_cast<int>(sql.length() + 1), &selectStmt, NULL);
+    while (1)
+    {
+        int state = sqlite3_step(selectStmt);
+        if (state == SQLITE_ROW)
+        {
+            std::string mac_addr = reinterpret_cast<const char*>(sqlite3_column_text(selectStmt, 0));
+            mac_addrs.push_back(mac_addr);
+            result = true;
+        }
+        else if (state == SQLITE_DONE)
+        {
+            break;
+        }
+        else
+        {
+            console_->error("Failed to read from database");
+            return false;
+        }
+    }
+    sqlite3_finalize(selectStmt);
+
+    console_->debug("- DataStore::GetRSSIDataStream");
+    return result;
```

Comment:
Unless there is a strong reason against this, I would suggest returning the `mac_addrs` instead.

Reply:
Done